### ISSUE TITLE
Conflict between object-shorthand-properties-first and object-literal-sort-key

### ISSUE THREAD

**Issue by brunolemos** [palantir/tslint#3606]
### Bug Report

- __TSLint version__: ^5.8.0
- __TypeScript version__: ^2.6.2
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
const a = 'a'
const c = 'c'
const obj = {
  a,
  b: 'b',
  c,
}
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:recommended", "tslint-config-airbnb", "tslint-react"]
}
```

#### Actual behavior

The rules conflict. There isn't a single field ordering that matches both rules, resulting on a non-fixable warning.

##### a, b, c
<img width="648" alt="object-shorthand-properties-first" src="https://user-images.githubusercontent.com/619186/34321975-ad5b722e-e802-11e7-8c52-22307b798801.png">

##### a, c, b
<img width="610" alt="object-literal-sort-keys" src="https://user-images.githubusercontent.com/619186/34321980-b82612f4-e802-11e7-8e4d-95a361901244.png">

#### Expected behavior

Not conflict.

_PS: I see this was mentioned on #2040 by @ajafff_

_Also I noticed `object-shorthand-properties-first` is from another repo but the fix makes more sense to be inside `object-literal-sort-key`_

**Comment by brunolemos** [comment]
I was about to make a PR but it seems there is already #2831 that fixes this

**Comment by brunolemos** [comment]
I created a PR anyway (#3607) because the existing one was outdated and included other features
